### DISCUSSION TITLE
DIFFICULTY DEPLOYING USING INJECTED METAMASK ON REMIX

### DISCUSSION THREAD

**Discussion by Lawwrenz** [Cyfrin/foundry-full-course-cu#3115] [Q&A]
Hello everyone, please ive  been having an issue deploying my smart contract using sepolia testnet on remix, i keep having the error below each time i try to deploy

"Please make sure that the current network is compatible with this evm version: paris. Otherwise any deployment will fail"

and i have manually gone to change it to paris yet its still not working , please how can i fix this issue?

**Comment by EngrPips** [comment]
Please show your remix page.

**Comment by Lawwrenz** [comment]
![Screenshot 2024-12-27 224316](https://github.com/user-attachments/assets/644c330d-487e-4b30-94bc-a24ae6fdb1d1)

**Reply by EngrPips** [reply]
What error do you get when you attempt deployment?

**Reply by Lawwrenz** [reply]
"Please make sure that the current network is compatible with this even version:Paris. otherwise any deployment will fail."

**Reply by EngrPips** [reply]
Are you able to click on the `Deploy` button? Please share me your code let me try on this end.

**Reply by Lawwrenz** [reply]
// SPDX-License-Identifier: MIT

pragma solidity 0.8.24; 

contract SimpleStorage {
    // This contrct allows us to store our favourite number
    // and other people's favourite number

    uint256  myFavouriteNumber;

    // uint256[] listOfFavouriteNumbers; // [0, 70, 90] 
    struct Person {
        uint256 favouriteNumber;
        string name;
        uint64  age;
    }  

    // dynamic array
    Person[] public listOfPeople;

    mapping(string => uint256) public nameToFavouriteNumber;                    

    function store(uint256 _favouriteNumber) public {
        myFavouriteNumber = _favouriteNumber;
    }

// view, pure
    function retireve() public view returns(uint256){
    return myFavouriteNumber;
    }

// calldata, memory, storage
    function addPerson(string memory _name, uint256 _favouriteNumber, uint64 _age) public{
        listOfPeople.push(Person(_favouriteNumber, _name, _age));
        nameToFavouriteNumber[_name] = _favouriteNumber;
    }
}

**Reply by Lawwrenz** [reply]
Each time i try to click on the deploy button,it is not responding

**Reply by EngrPips** [reply]
```js
// SPDX-License-Identifier: MIT

pragma solidity 0.8.24;

contract SimpleStorage {
// This contrct allows us to store our favourite number
// and other people's favourite number

uint256  myFavouriteNumber;

// uint256[] listOfFavouriteNumbers; // [0, 70, 90] 
struct Person {
    uint256 favouriteNumber;
    string name;
    uint64  age;
}  

// dynamic array
Person[] public listOfPeople;

mapping(string => uint256) public nameToFavouriteNumber;                    

function store(uint256 _favouriteNumber) public {
    myFavouriteNumber = _favouriteNumber;
}
// view, pure
function retireve() public view returns(uint256){
return myFavouriteNumber;
}

// calldata, memory, storage
function addPerson(string memory _name, uint256 _favouriteNumber, uint64 _age) public{
listOfPeople.push(Person(_favouriteNumber, _name, _age));
nameToFavouriteNumber[_name] = _favouriteNumber;
}
}
```

**Reply by EngrPips** [reply]
> Each time i try to click on the deploy button,it is not responding

Have you tried refreshing `remix`?

**Reply by EngrPips** [reply]
> ```js
> // SPDX-License-Identifier: MIT
> 
> pragma solidity 0.8.24;
> 
> contract SimpleStorage {
> // This contrct allows us to store our favourite number
> // and other people's favourite number
> 
> uint256  myFavouriteNumber;
> 
> // uint256[] listOfFavouriteNumbers; // [0, 70, 90] 
> struct Person {
>     uint256 favouriteNumber;
>     string name;
>     uint64  age;
> }  
> 
> // dynamic array
> Person[] public listOfPeople;
> 
> mapping(string => uint256) public nameToFavouriteNumber;                    
> 
> function store(uint256 _favouriteNumber) public {
>     myFavouriteNumber = _favouriteNumber;
> }
> // view, pure
> function retireve() public view returns(uint256){
> return myFavouriteNumber;
> }
> 
> // calldata, memory, storage
> function addPerson(string memory _name, uint256 _favouriteNumber, uint64 _age) public{
> listOfPeople.push(Person(_favouriteNumber, _name, _age));
> nameToFavouriteNumber[_name] = _favouriteNumber;
> }
> }
> ```

I will try on my end and see.

**Reply by EngrPips** [reply]
Hey, things work fine for me.

![Screenshot 2024-12-29 at 05 33 16](https://github.com/user-attachments/assets/6a749146-c8fe-457c-99bb-5fa35bb81bd0)

**Reply by EngrPips** [reply]
I think you should be able to click on the `Deploy` button

**Reply by Lawwrenz** [reply]
I did refreshed severally

**Comment by H-code9** [comment]
I hope you've gotten this issue solved @LawrenceEzealor ? kindly implement the suggested tips from @EngrPips .

Best regards.

**Reply by Lawwrenz** [reply]
Im not sure if this is a problem with the code of remix, even when I finish with my FUNDME contract and trying to deploy on testnet, still the same issue, It was preventing me from me from deploying and my injected metamask isn't popping up and also not evening showing my metamask address . it says it should be Paris, which is on Paris yet still not working .

**Reply by Tobizeph** [reply]
check for "wallet connect" in that option list. Use that instead of "injected provider - metamask". Mine worked that way

**Reply by Lawwrenz** [reply]
I did tried wallet connect, clicked on it severally, not responding unfortunately .

**Reply by EngrPips** [reply]
Maybe restart your machine and also clear your browser cache and then retry.

**Comment by Tobizeph** [comment]
Hello everyone, I also face the same issue deploying my smart contract using sepolia testnet on remix, it's not showing my sepolia eth balance and I can't press the deploy button. How can I get it done please?

**Reply by EngrPips** [reply]
Show your Remix workspace.

**Comment by Lawwrenz** [comment]
I was also going to mention that ive also tried severally to get the zkSync plugin on remix, i cant seem to find it also

**Reply by EngrPips** [reply]
This will usually happen due to an unstable internet connection.
